client/signup/steps/site-or-domain/index.jsx

```diff
@@ -18,21 +18,27 @@ import { getCurrentUserId } from 'state/current-user/selectors';
 import DomainImage from 'signup/steps/design-type-with-store/domain-image';
 import NewSiteImage from 'signup/steps/design-type-with-store/new-site-image';
 import ExistingSite from 'signup/steps/design-type-with-store/existing-site';
-import { externalRedirect } from 'lib/route/path';
 import NavigationLink from 'signup/navigation-link';
+import QueryProductsList from 'components/data/query-products-list';
+import { getProductsList } from 'state/products-list/selectors';
 
 class SiteOrDomain extends Component {
-	componentWillMount() {
-		if ( ! this.getDomainName() ) {
-			// /domains domain search is an external application to calypso,
-			// therefor a full redirect required:
-			externalRedirect( '/domains' );
+	getDomainProductSlug( domain ) {
+		const domainParts = domain.split( '.' );
+		const tld = domainParts.slice( 1 ).join( '.' );
```

Comment:
We [have `getTld`](https://github.com/Automattic/wp-calypso/blob/master/client/lib/domains/index.js#L139) that should be helpful here :)